### PR TITLE
Add writeConnector to file utils

### PR DESCRIPTION
**Pull request by hlshen**

### CODE REVIEW

File: src/dataconnect/fileUtils.ts
Comment (joehan) [inline src/dataconnect/fileUtils.ts:140]: Npm run format on firebase-tools

**Review by joehan — COMMENTED** [review]
I think we bumped into each other on this one and https://github.com/firebase/firebase-tools/pull/7522/files - can you double check that the logic here is still up to date after merging master?

File: src/dataconnect/fileUtils.ts
Comment (joehan) [inline src/dataconnect/fileUtils.ts:157]: Edge case to check for when using this function - If i don't provide an app folder, what does this write to connector.yaml? Does it leave an empty `packageJsonDir:`?

File: src/dataconnect/fileUtils.ts
Comment (joehan) [inline src/dataconnect/fileUtils.ts:172]: Nit: Unnecessary newline?

**Review by joehan — APPROVED** [review]
LGTM, thanks for adding tests! One case to call out

File: src/dataconnect/fileUtils.ts
Comment (hlshen) [reply src/dataconnect/fileUtils.ts:157]: hrm, are you suggesting we check if appFolder is empty? outputDir is also being generated from appFolder

Comment (joehan) [reply src/dataconnect/fileUtils.ts:157]: This is looking forward to follow up PRs that start using this function, so no need to block on this: 'packageJsonDir' is optional, to support users that use package managers other than npm. We should make sure that the UI makes it easy for users to omit it, and we should make sure that the final `connector.yaml` doesn't have an empty `packageJsonDir` field:

```
generate:
 webSdk:
    outputDir: ../../dir
    package: connectors.default
    packageJsonDir:
```

Comment (hlshen) [reply src/dataconnect/fileUtils.ts:157]: ah, this would be for CLI init flow no? The auto detect doesn't have any web indicators aside from npm indiciators.

**Pull request merged**
